extensions/default/src/DicomLocalDataSource/index.js

```diff
@@ -153,14 +153,33 @@ function createDicomLocalApi(dicomLocalConfig) {
               } = instance;
 
               instance.imageId = imageId;
-
-              // Add imageId specific mapping to this data as the URL isn't necessarily WADO-URI.
-              metadataProvider.addImageIdToUIDs(imageId, {
-                StudyInstanceUID,
-                SeriesInstanceUID,
-                SOPInstanceUID,
-                frameIndex: isMultiframe ? index : 1,
-              });
+              if (isMultiframe) {
```

Comment:
Please remove the if (isMultiframe) and just do the loop to:
const numberOfFrames = instance.NumberOfFrames || 1
That way it is consistently calling the same code and is likely to be more robust.  The getImageIdsForInstance should work for single frames to on frame===1

Reply:
That will make frame 1 of single frame images have a consistent URL with the other images, and the frame image id will not be the same as the overall imageId for any image, which tends to fix issues around using the wrong imageid as they will be caught consistently.